src/screens/Dashboard.js

```diff
@@ -0,0 +1,198 @@
+        </Typography>
+      ),
+    };
+    return newCell;
+  }
+
+  render() {
+    const { classes, list } = this.props;
+    const columns = [
+      {
+        Header: "",
+        Cell: () => (
+          <Avatar
+            className={classes.smallAvatar}
+            src={`https://randomuser.me/api/portraits/men/${Math.floor(Math.random() * 100)}.jpg`}
+            alt="avatar placeholder"
+          />
+        ),
+        filterable: false,
+      },
+      {
+        id: "fullName",
+        Header: "Full name",
+        accessor: d => `${d.first_name} ${d.last_name}`,
+        Cell: (props) => {
+          const userData = [
+            props.value,
+            props.original,
+          ];
+
+          return (
+            <TextButton>
+              <Typography
+                onClick={() => {
+                  this.setState({
+                    selectedUser: userData,
+                    openModal: true,
+                  });
+                }}
+                variant="body1"
+              >
+                {props.value}
+              </Typography>
+            </TextButton>
+          );
+        },
+        filterMethod: (filter, row) => matchSorter([row[filter.id]], filter.value).length !== 0,
+      },
+      this.generateColumn("first_name", false, false),
+      this.generateColumn("last_name", false, false),
+      this.generateColumn("birth_date"),
+      this.generateColumn("phone_number"),
+    ];
+
+    return (
+      <div className="dashboard-grid">
+        <div style={{ gridArea: "nav" }} />
+        <AppBar className={classes.topbar}>
+          <Toolbar className={classes.toolbar}>
+            <Button
+              onClick={() => this.refreshList()}
+              color="secondary"
+              className={classes.refreshBtn}
+            >
+              <Refresh />
+              {" "}
+              Refresh list
+            </Button>
+            <TextField
+              id="fullName"
+              type="text"
+              className={`${classes.TextField} name-filter`}
+              autoComplete="off"
+              onChange={event => (
+                this.reactTable.current.filterColumn(columns[1], event.target.value)
+              )}
+              InputProps={{
+                startAdornment: (
+                  <InputAdornment position="start">
+                    <Search />
+                  </InputAdornment>
+                ),
+                className: classes.fullNameFilter,
+              }}
+              placeholder="Search by name..."
+              variant="outlined"
+            />
+          </Toolbar>
+        </AppBar>
+        <main className={classes.rtWrapper}>
+          <ModalCard
+            modalData={this.state.selectedUser}
+            openModal={this.state.openModal}
+            onClose={() => {
+              this.setState({ openModal: false });
+              setTimeout(() => {
+                this.setState(defaultState.selectedUser);
```

Comment:
i don't think you need to clean the data here, and even if you do, why not using the above setState, why need setTimeout